### ISSUE TITLE
All in one in version 4.2 not working properly.

### ISSUE THREAD

**Issue by pdubois** [Alfresco/alfresco-sdk#422]
First problem encountered is when following  https://docs.google.com/document/d/1p5QR1Q8pVvMT9fh_aID5nraD5NP6ZrU1ztcgE_zHR-4/pub#h.sgeg9yx1fo8y section “Using Alfresco 4.2 » for all in one.

There is no file aio/src/test/properties/local/alfresco-global.properties but 
aio/aio-platform-jar/src/main/resources/alfresco/module/aio-platform-jar/alfresco-global.properties

Question is should « index.subsystem.name=solr » be set in previous file instead?

Second issue is when starting the project using 
mvn clean install -DskipTests=true alfresco:run

It does not seems to be able to connect to DB.

See : attached error.log in attached zip 
[error.log.zip](https://github.com/Alfresco/alfresco-sdk/files/541762/error.log.zip)

**Comment by jgoldhammer** [comment]
The same for me in a new repo-amp project for alfresco 4.2.f!

Steps to reproduce:
* Generate a new project with following command
`mvn archetype:generate -DarchetypeCatalog=https://artifacts.alfresco.com/nexus/content/groups/public/archetype-catalog.xml -Dfilter=org.alfresco.maven.archetype:`

* Change pom.xml according the docs
  * remove the alfresco-share-services
  * change version to 4.2.f

* Execute run.sh

Result:
Alfresco does not start with the following exception.

```
Caused by: org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'repoSqlSessionFactory' defined in class path resource [alfresco/ibatis/ibatis-context.xml]: Invocation of init method failed; nested exception is org.springframework.core.NestedIOException: Failed to parse config resource: class path resource [alfresco/ibatis/alfresco-SqlMapConfig.xml]; nested exception is org.apache.ibatis.builder.BuilderException: Error parsing SQL Mapper Configuration. Cause: org.apache.ibatis.builder.BuilderException: Failed to get resource: alfresco/ibatis/#resource.dialect#/activities-insert-SqlMap.xml
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.initializeBean(AbstractAutowireCapableBeanFactory.java:1420)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:519)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:456)
	at org.springframework.beans.factory.support.AbstractBeanFactory$1.getObject(AbstractBeanFactory.java:291)
	at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:222)
	at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:288)
	at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:190)
	at org.springframework.beans.factory.support.BeanDefinitionValueResolver.resolveReference(BeanDefinitionValueResolver.java:322)
	... 57 more
Caused by: org.springframework.core.NestedIOException: Failed to parse config resource: class path resource [alfresco/ibatis/alfresco-SqlMapConfig.xml]; nested exception is org.apache.ibatis.builder.BuilderException: Error parsing SQL Mapper Configuration. Cause: org.apache.ibatis.builder.BuilderException: Failed to get resource: alfresco/ibatis/#resource.dialect#/activities-insert-SqlMap.xml
	at org.alfresco.ibatis.HierarchicalSqlSessionFactoryBean.buildSqlSessionFactory(HierarchicalSqlSessionFactoryBean.java:248)
	at org.alfresco.ibatis.HierarchicalSqlSessionFactoryBean.afterPropertiesSet(HierarchicalSqlSessionFactoryBean.java:98)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.invokeInitMethods(AbstractAutowireCapableBeanFactory.java:1477)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.initializeBean(AbstractAutowireCapableBeanFactory.java:1417)
	... 64 more
Caused by: org.apache.ibatis.builder.BuilderException: Error parsing SQL Mapper Configuration. Cause: org.apache.ibatis.builder.BuilderException: Failed to get resource: alfresco/ibatis/#resource.dialect#/activities-insert-SqlMap.xml
	at org.alfresco.ibatis.HierarchicalXMLConfigBuilder.parseConfiguration(HierarchicalXMLConfigBuilder.java:128)
	at org.alfresco.ibatis.HierarchicalXMLConfigBuilder.parse(HierarchicalXMLConfigBuilder.java:110)
	at org.alfresco.ibatis.HierarchicalSqlSessionFactoryBean.buildSqlSessionFactory(HierarchicalSqlSessionFactoryBean.java:246)
	... 67 more
Caused by: org.apache.ibatis.builder.BuilderException: Failed to get resource: alfresco/ibatis/#resource.dialect#/activities-insert-SqlMap.xml
	at org.alfresco.ibatis.HierarchicalXMLConfigBuilder.mapperElement(HierarchicalXMLConfigBuilder.java:306)
	at org.alfresco.ibatis.HierarchicalXMLConfigBuilder.parseConfiguration(HierarchicalXMLConfigBuilder.java:124)
	... 69 more
Nov 06, 2016 8:18:20 PM org.apache.catalina.core.StandardContext listenerStart
```

**Comment by ohej** [comment]
Fixed in 245a51402d1a428041e6f125d7c7766286827067

**Comment by ohej** [comment]
Also to add more color to it: Index subsystem needed to be `solr` vs `solr4`. Solr 1.4 had multiple issues which were fixed in 245a51402d1a428041e6f125d7c7766286827067 as well. On top of that the Solr war file needed to be exploded and `web.xml` changed. This works all out of the box now
